bundles/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/handler/HomematicThingHandler.java

```diff
@@ -384,8 +384,10 @@ protected void updateDatapointState(HmDatapoint dp) {
     private void updateChannelState(final HmDatapoint dp, Channel channel)
             throws IOException, GatewayNotAvailableException, ConverterException {
         if (dp.isTrigger()) {
-            if (dp.getValue() != null) {
-                triggerChannel(channel.getUID(), dp.getValue() == null ? "" : dp.getValue().toString());
+            final Object value = dp.getValue();
+            final Object prevValue = dp.getPreviousValue();
```

Comment:
A separate varaible vor prevValue is not necessary, because you only use in the next statement.